pyproject.toml

```diff
@@ -285,10 +285,21 @@ asyncio_mode = "auto"
 asyncio_default_fixture_loop_scope = "class"
 
 [tool.mypy]
+python_version = "3.9"
```

Comment:
I encountered CI errors due to accidentally using typing constructs that are not supported by Python 3.9. Therefore, I added this option to the Mypy configuration so that `hatch run test:types` catches such issues early.